specification/securityinsights/resource-manager/Microsoft.SecurityInsights/stable/2024-09-01/dataConnectorDefinitions.json

```diff
@@ -0,0 +1,706 @@
+          }
+        ]
+      }
+    },
+    "DataConnectorDefinitionArmCollectionWrapper": {
+      "type": "object",
+      "description": "Encapsulate the data connector definition object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DataConnectorDefinition"
+          }
+        },
+        "nextLink": {
+          "type": "string"
+        }
+      }
+    },
+    "CustomizableConnectorDefinition": {
+      "description": "Connector definition for kind 'Customizable'.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/DataConnectorDefinition"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/CustomizableConnectorDefinitionProperties",
+          "description": "Customizable properties.",
+          "x-ms-client-flatten": true
+        }
+      },
+      "x-ms-discriminator-value": "Customizable"
+    },
+    "CustomizableConnectorDefinitionProperties": {
+      "description": "The UiConfig for 'Customizable' connector definition kind.",
+      "type": "object",
+      "required": [
+        "connectorUiConfig"
+      ],
+      "properties": {
+        "createdTimeUtc": {
+          "format": "date-time",
+          "description": "Gets or sets the connector definition created date in UTC format.",
+          "type": "string"
+        },
+        "lastModifiedUtc": {
+          "format": "date-time",
+          "description": "Gets or sets the connector definition last modified date in UTC format.",
+          "type": "string"
+        },
+        "connectorUiConfig": {
+          "$ref": "#/definitions/CustomizableConnectorUiConfig"
+        },
+        "connectionsConfig": {
+          "$ref": "#/definitions/CustomizableConnectionsConfig"
+        }
+      }
+    },
+    "CustomizableConnectionsConfig": {
+      "description": "The UiConfig for 'Customizable' connector definition kind.",
+      "required": [
+        "templateSpecVersion",
+        "templateSpecName"
+      ],
+      "type": "object",
+      "properties": {
+        "templateSpecName": {
+          "description": "Gets or sets the template name. The template includes ARM templates that can be created by the connector, usually it will be the dataConnectors ARM templates.",
+          "type": "string"
+        },
+        "templateSpecVersion": {
+          "description": "Gets or sets the template version.",
+          "type": "string"
+        }
+      }
+    },
+    "CustomizableConnectorUiConfig": {
+      "description": "The UiConfig for 'Customizable' connector definition kind.",
+      "required": [
+        "dataTypes",
+        "descriptionMarkdown",
+        "graphQueries",
+        "instructionSteps",
+        "permissions",
+        "publisher",
+        "title",
+        "connectivityCriteria"
+      ],
+      "type": "object",
+      "properties": {
+        "id": {
```

Comment:
>id [](http://example.com/codeflow?start=9&length=2)

Don't duplicate top-level ARM properties. If this is an Azure resourceId, you should use the name resourceId or connectorResourceId, and add format: arm-id. if it's something else, clarify what it is in the description (e.g. number, name, guid, etc.).